statistics/histogram.go

```diff
@@ -1441,7 +1441,28 @@ func (idx *Index) GetRowCount(sctx sessionctx.Context, coll *HistColl, indexRang
 				return 0, err
 			}
 			if expBackoffSuccess {
-				totalCount += expBackoffSel * idx.TotalRowCount()
+				expBackoffCnt := expBackoffSel * idx.TotalRowCount()
+
+				upperLimit := expBackoffCnt
+				// Use the multi-column stats to calculate the max possible row count of [l, r)
+				if idx.Len() > 0 {
+					_, lowerBkt, _, _ := idx.locateBucket(l)
+					_, upperBkt, _, _ := idx.locateBucket(r)
```

Comment:
Do we need to skip the cases that `lowerBkt == upperBkt`?

Reply:
I think we don't need to. Because we use the bucket before where `l` exists as the lower bound.

Reply:
I added some comments to clarify this.